builder/builder.go

```diff
@@ -128,6 +128,14 @@ type Docker interface {
 	// Release releases a list of images that were retained for the time of a build.
 	// TODO: remove
 	Release(sessionID string, activeImages []string)
+	// Kill stops the container execution abruptly.
+	Kill(c *daemon.Container) error
+	// Mount mounts the root filesystem for the container.
+	Mount(c *daemon.Container) error
+	// Unmount unmounts the root filesystem for the container.
+	Unmount(c *daemon.Container) error
+	// Start starts a new container
+	Start(c *daemon.Container) error
```

Comment:
I think all of these should take a ContainerID reference, and do the lookup inside of the daemon instead of taking a daemon struct reference. Otherwise daemon.Container needs to be a struct in a pkg for everything to reference.